usrsctplib/user_socket.c

```diff
@@ -1564,8 +1579,9 @@ sowakeup(struct socket *so, struct sockbuf *sb)
 #endif
 	}
 	SOCKBUF_UNLOCK(sb);
-	/*__Userspace__ what todo about so_upcall?*/
 
+	if ((sb->sb_flags & SB_UPCALL) && so->so_upcall != NULL)
```

Comment:
Why is this change needed?

Reply:
I do really not know why this change.

Reply:
@tuexen, as said, this is an extract from #38. I've tried to extract merely the parts related to the single thread. We've tested quite intensively and it's working as per our usage. I cannot determine the reason of such change not whether it is necessary or not.

Please, feel free to edit the code if needed.

Reply:
I think it is not needed. We currently call the upcall after we processed the triggering event. So please remove it.

Reply:
Done in https://github.com/versatica/usrsctp/commit/3ca5f0b1047e9f72a78aa61e8ac6fb224942bc98.